datasource/virtual_machine/data.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type Config,Tag,DatasourceOutput
+package virtual_machine
+
+import (
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/hashicorp/packer-plugin-sdk/common"
+	"github.com/hashicorp/packer-plugin-sdk/hcl2helper"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/template/config"
+	vsCommon "github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common"
+	"github.com/pkg/errors"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// Example of multiple vm_tags blocks in HCL format:
+// ```
+//
+//	vm_tags {
+//	  category = "team"
+//	  name = "operations"
+//	}
+//	vm_tags {
+//	  category = "SLA"
+//	  name = "gold"
+//	}
+//
+// ```
```

Comment:
```suggestion
// HCL Example:
//
// ```hcl
//	vm_tags {
//	  category = "team"
//	  name = "operations"
//	}
//	vm_tags {
//	  category = "sla"
//	  name = "gold"
//	}
// ```
```